### PR TITLE
build nightly docker images

### PR DESCRIPTION
**Pull request by techknowlogick**
followup of https://github.com/go-gitea/gitea/pull/25308 this time to build & push nightly docker images

### CODE REVIEW

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
> +30 −306

Very nice and shows how needlessly verbose drone actually is 😆.

**Comment by techknowlogick** [comment]
> Very nice and shows how needlessly verbose drone actually is

to be fair to drone, this is running the multi-arch build using qemu, and doing the manifest as well. For drone we were using native runners so it had to be separated (if we did the same with GHA then there'd be much more yaml too). So a bit of apples and oranges here.

File: .github/workflows/release-nightly.yml [hunk 1]
```diff
@@ -55,5 +55,33 @@ jobs:
...
+      - name: Get cleaned branch name
+        id: clean_name
+        run: |
+          # if main then say nightly otherwise cleanup name
+          if [ "${{ github.ref }}" = "refs/heads/main" ]; then
+            echo "branch=nightly" >> "$GITHUB_OUTPUT"
+            exit 0
+          fi
+          REF_NAME=$(echo "${{ github.ref }}" | sed -e 's/refs\/heads\///' -e 's/refs\/tags\///' -e 's/release\/v//')
+          echo "branch=${REF_NAME}" >> "$GITHUB_OUTPUT"
```
Comment (techknowlogick) [inline .github/workflows/release-nightly.yml]:
```suggestion
          echo "branch=${REF_NAME}-nightly" >> "$GITHUB_OUTPUT"
```

File: .github/workflows/release-nightly.yml [hunk 2]
```diff
@@ -55,5 +55,33 @@ jobs:
...
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: build rootful docker image
+        uses: docker/build-push-action@v4
+        with:
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: gitea/gitea:${{ steps.clean_name.outputs.branch }}-nightly
```
Comment (techknowlogick) [inline .github/workflows/release-nightly.yml]:
```suggestion
          tags: gitea/gitea:${{ steps.clean_name.outputs.branch }}
```

File: .github/workflows/release-nightly.yml [hunk 3]
```diff
@@ -55,5 +55,33 @@ jobs:
...
+      - name: build rootless docker image
+        uses: docker/build-push-action@v4
+        with:
+          platforms: linux/amd64,linux/arm64
+          push: true
+          file: Dockerfile.rootless
+          tags: gitea/gitea:${{ steps.clean_name.outputs.branch }}-nightly-rootless
```
Comment (techknowlogick) [inline .github/workflows/release-nightly.yml]:
```suggestion
          tags: gitea/gitea:${{ steps.clean_name.outputs.branch }}-rootless
```

**Review by techknowlogick — COMMENTED** [review]
added these changes so that it doesn't result in an image named nightly-nightly

**Review by lunny — APPROVED** [review]

**Pull request merged**